src/com/facebook/buck/util/DefaultProcessExecutor.java

```diff
@@ -217,6 +217,13 @@ public Result waitForLaunchedProcessWithTimeout(
     );
   }
 
+  @Override
+  public boolean isAlive(LaunchedProcess launchedProcess) {
+    Preconditions.checkState(launchedProcess instanceof LaunchedProcessImpl);
+    final Process process = ((LaunchedProcessImpl) launchedProcess).process;
```

Comment:
Rather than accessing the field after the instanceof check add a `isAlive` method to `LaunchedProcess`.

It's possible the method won't be necessary on the executor at that point.